syntax = "proto3";
package processing;
// Define the Point3 message with three floating-point fields

message Point3 {
    float x = 1;
    float y = 2;
    float z = 3;
}

// Define the Numeric message with a single floating-point field
message Numeric {
    float value = 1;
}

// Define the ImageRequest message with image size field
message ImageRequest {
    bytes image_data = 1; // image data in bytes
}

message ImageResponse {
    bytes resized_image_data = 1; //Resized Image in bytes
}

// Define the ProcessingServices service with a computeSum method
service  ProcessingServices {
    // Method to compute the sum of three numbers
    rpc computeSum(Point3) returns (Numeric) {}

    // Method to resize an image
    rpc resizeImage(ImageRequest) returns (ImageResponse) {}
}
